### PR TITLE
add command to generate a csv of sync log profile data

### PR DESCRIPTION
**Pull request by czue**
http://manage.dimagi.com/default.asp?160836

note that this doesn't work until https://github.com/dimagi/commcare-hq/pull/5931 is merged

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/management/commands/sync_log_profile.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    def handle(self, filename, *args, **kwargs):
+        database = SyncLog.get_db()
+        all_sync_log_ids = [
+            row['id'] for row in
+            database.view('phone/sync_logs_by_user', reduce=False, include_docs=False)
+        ]
+        total_count = len(all_sync_log_ids)
+
+        headers = [
+            'domain', 'user', 'cases',  'dependent cases', 'total cases', 'initial', 'duration',
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/management/commands/sync_log_profile.py]: E241 multiple spaces after ','

**Comment by czue** [comment]
@snopoke @dannyroberts

File: corehq/ex-submodules/casexml/apps/phone/management/commands/sync_log_profile.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            writer.writerow(
+                headers
+            )
+            for i, sync_log_dict in enumerate(iter_docs(database, all_sync_log_ids, 500)):
+                duration = sync_log_dict.get('duration')
+                cases = len(sync_log_dict.get('cases_on_phone', []))
+                dependent_cases = len(sync_log_dict.get('dependent_cases_on_phone', []))
+                total_cases = cases + dependent_cases
+                if duration and total_cases:
+                    average_time = float(duration) * 1000 / float(total_cases) if total_cases else 0
```
Comment (millerdev) [inline corehq/ex-submodules/casexml/apps/phone/management/commands/sync_log_profile.py]: `if total_cases else 0` seems unnecessary as it must evaluate to true on the line above.

Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/management/commands/sync_log_profile.py]: whoops you're right. this was before i pulled that up to the if.

Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/management/commands/sync_log_profile.py]: https://github.com/dimagi/commcare-hq/pull/5939

**Pull request merged**